src/MultiLocationEncoder.sol

```diff
@@ -1,3 +1,54 @@
+// SPDX-License-Identifier: Apache-2.0
 pragma solidity ^0.8.13;
 
-contract MultiLocationEncoder {}
+import "./XCMEnums.sol";
+import "./utils/ScaleCodec.sol";
+import "./utils/CompactTypes.sol";
+
+contract MultiLocationEncoder {
+
+    function encodeMultiLocationSameConsensusParachain(
+        uint8 parents,
+        uint256 parachainId
+    )
+    public pure returns (bytes memory) {
+        XCMEnums.XcmV3Junctions interiorJunctions = XCMEnums.XcmV3Junctions.X1;
+        XCMEnums.XcmV3Junction firstJunction = XCMEnums.XcmV3Junction.Parachain;
+        return abi.encodePacked(
```

Comment:
Should rather use [byte.concat](https://docs.soliditylang.org/en/v0.8.17/types.html#the-functions-bytes-concat-and-string-concat), which was introduced for this purpose. The former `abi.encodePacked` is [discouraged](https://github.com/ethereum/solidity/issues/10903) for concatenating bytes.